Assets/PoisonMushroom.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PoisonMushroom : MonoBehaviour
+{
+    //Mushroom stats
+    public int health; //Mushroom health
+    public int damageNum; //amount of damage applied to Mushroom health
+    public float lifeTimer; //timer for destroying the Mushroom
+    private float generateTimer; //timer for generating the clouds
+    private BoxCollider collider;
+    private MeshRenderer mesh;
+
+    //Generating clouds using Particle System
+    private ParticleSystem cloud;
+    public float timeGenerate; //duration for generating the clouds
+
+    void Start()
+    {
+        collider = GetComponent<BoxCollider>();
+        mesh = GetComponent<MeshRenderer>();
+        cloud = GetComponent<ParticleSystem>();
+        generateTimer = timeGenerate;
+    }
+
+    void Update()
+    {
+        if (generateTimer > 0)
+        {
+            generateTimer -= Time.deltaTime;
+        }
+        else if (generateTimer == 0)
+        {
+            cloud.Stop();
+            cloud.Clear();
+        }
+        else if (generateTimer <= 0)
+        {
+            GeneratePoisonCloud();
+            generateTimer = timeGenerate;
+        }
+
+        //Start death sequence
+        if (health <= 0)
+        {
+            lifeTimer -= Time.deltaTime;
+            DestroyMushroom();
+        }
+        if (lifeTimer < 0)
+        {
+            Debug.Log("Destroy Mushroom object");
+            Destroy(gameObject);
+        }
+    }
+
+    public void GeneratePoisonCloud()
+    {
+        cloud.Simulate(0.0f, true, true); //reset particle system timer
+        cloud.Play();
+        Debug.Log("Generate clouds");
+    }
+
+    public void TakeDamage(int damage)
+    {
+        health -= damage;
+    }
+
+    //If player collides with the Mushroom, then destroy this gameObject
+    void OnCollisionEnter(Collision collision)
```

Comment:
The OnCollisionEnter method should be removed since the collision with the bullet and player are handled by the bullet and PlayerHealth Scripts